src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -282,6 +282,34 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
+  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = workflowLoggerWithCall(backendCall)
+    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
+      Try(gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log))) match {
+        case Success(attempts) if attempts.forall(_.isSuccess) =>
+          postProcess(backendCall) match {
+            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
+            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
+              // TODO: What can we return here to retry this operation?
+              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
+              log.error(error, ex)
+              FailedExecutionHandle(new Throwable(error, ex))
+            case Failure(ex) => FailedExecutionHandle(ex)
+          }
+        case Success(attempts) =>
+          TryUtil.sequence(attempts.toSeq) match {
```

Comment:
With the disclaimer that this is green in IntelliJ but I didn't actually run it, and also that the breadth of the changes makes me somewhat reluctant to gate this PR on actually doing this:

`GoogleCloudStorage.copyPrefix` could become:

```scala
  def copyPrefix(from: String, to: String, logger: Option[WorkflowLogger] = None): ValidationNel[String, Seq[StorageObject]] = {
    val sourceToDestinationMap = listContents(from) map { x =>
      GcsPath(x) -> GcsPath(x.replaceAll(s"^$from", to))
    }
    val nels = sourceToDestinationMap map { case (source, dest) =>
      // TODO: This next call can throw an exception, like if the objectName is an empty string.
      val storageObject = client.objects.get(source.bucket, source.objectName).execute
      Try(client.objects.copy(source.bucket, source.objectName, dest.bucket, dest.objectName, storageObject).execute) match {
        case Failure(ex) =>
          logger.foreach(_.error(s"Could not copy $source -> $dest", ex))
          s"$source -> $dest".failureNel
        case Success(obj) =>
          logger.foreach(_.info(s"Copied $source -> $dest"))
          obj.successNel
      }
    }
    val validationNel = nels.toList.sequence[({type λ[a] = ValidationNel[String, a]})#λ, StorageObject]
    // With a leftMap, prepend an error message to the concatenated error values if there are error values.
    // This turns the ValidationNel into a Validation, force it back to a ValidationNel with toValidationNel.
    validationNel.leftMap("Could not copy files: " + _.list.mkString(", ")).toValidationNel
  }
```

Then this reduces to:

```scala
  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
    val log = workflowLoggerWithCall(backendCall)
    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
      gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log)) match {
        case scalaz.Success(attempts) =>
          postProcess(backendCall) match {
            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
              // TODO: What can we return here to retry this operation?
              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
              log.error(error, ex)
              FailedExecutionHandle(new Throwable(error, ex))
            case Failure(ex) => FailedExecutionHandle(ex)
          }
        case _ => 
          // All the failures were already logged.
          FailedExecutionHandle(new Throwable("Error copying cached files"))
      }
    }
  }
```

There's more TryUtil in there that could be squashed out.